### PR TITLE
Add some notes about tests and documentation

### PR DESCRIPTION
**Pull request by Soullivaneuh**
About `@cover` usage, this was discussed on another PR, we are following the Symfony way.

If someone can retrieve the link... @greg0ire maybe?

### CODE REVIEW

**Comment by greg0ire** [comment]
I'll look on slack

**Comment by greg0ire** [comment]
https://github.com/sonata-project/SonataAdminBundle/pull/3809#discussion_r63182577

File: project/CONTRIBUTING.md [hunk 1]
```diff
@@ -64,9 +66,27 @@ and run this command before committing your modifications:
...
+
+The documentation is mostly written with the `rst` format. You can test the doc rendering with the `make docs` command.
+
+If your PR contains a new feature, you must add documentation for it.
+
+#### The tests
+
+If your PR contains a fix, tests should be added to prove the bug.
+
+If your PR contains an addition, a new feature, this one have to be fully covered by tests.
```
Comment (greg0ire) [inline project/CONTRIBUTING.md]: "has to"

File: project/CONTRIBUTING.md [hunk 2]
```diff
@@ -64,9 +66,27 @@ and run this command before committing your modifications:
...
+
+Some rules have to be respected about the test:
+
+* Annotations about coverage like `@covers`, `@coversDefaultClass` and `@coversNothing` are prohibited.
+* All test methods should be prefixed by `test`. Example: `public function testItReturnsNull()`.
```
Comment (greg0ire) [inline project/CONTRIBUTING.md]: "as opposed to using the `@test` annotation"

File: project/CONTRIBUTING.md [hunk 3]
```diff
@@ -64,9 +66,34 @@ and run this command before committing your modifications:
...
+
+Some rules have to be respected about the test:
+
+* Annotations about coverage are prohibited. This concerns:
+  * `@covers`
+  * `@coversDefaultClass`
+  * `@coversNothing`
+  * `@codeCoverageIgnore`
+  * `@codeCoverageIgnoreStart`
+  * `codeCoverageIgnoreEnd`
```
Comment (core23) [inline project/CONTRIBUTING.md]: `@` is missing

Comment (Soullivaneuh) [inline project/CONTRIBUTING.md]: Erf, too bad. Will fix it, thanks!

**Pull request merged**